lib/ansible/cli/doc.py

```diff
@@ -793,15 +793,20 @@ def namespace_from_plugin_filepath(filepath, plugin_name, basedir):
     @staticmethod
     def _get_plugin_doc(plugin, plugin_type, loader, search_paths):
         # if the plugin lives in a non-python file (eg, win_X.ps1), require the corresponding python file for docs
-        result = loader.find_plugin_with_context(plugin, mod_type='.py', ignore_deprecated=True, check_aliases=True)
-        if not result.resolved:
-            raise PluginNotFound('%s was not found in %s' % (plugin, search_paths))
+        for ext in ('.py', '.yml', '.yaml'):
+            result = loader.find_plugin_with_context(plugin, mod_type=ext, ignore_deprecated=True, check_aliases=True)
```

Comment:
This is problematic, since you may end up picking up an unrelated doc file from a completely different location due to all the internal loader path shuffle/fallback; also problematic for redirects. Suggest using the loader to load the plugin itself, then use the resolved path from the context and look for an adjoining `.yml` right next to it (or if the resolved one isn't `.py`, look for that first. That should minimize cases where you pick up an unrelated doc from somewhere else on the plugin search path.

Reply:
known issue , (see irc discussion with felix) but requires loader changes and this enabled the 'demo' of the existing files (happy path), the plan is changing the jinja2 plugin loader to return file path (even when not matching name), then use basename there